core-rust-bridge/src/main/java/com/radixdlt/statemanager/DatabaseTypeConfig.java

```diff
@@ -68,23 +68,23 @@
 import com.radixdlt.sbor.codec.EnumCodec;
 
 /** REv2 Database configuration options */
-public sealed interface REv2DatabaseConfig {
+public sealed interface DatabaseTypeConfig {
   static void registerCodec(CodecMap codecMap) {
     codecMap.register(
-        REv2DatabaseConfig.class,
-        codecs -> EnumCodec.fromPermittedRecordSubclasses(REv2DatabaseConfig.class, codecs));
+        DatabaseTypeConfig.class,
+        codecs -> EnumCodec.fromPermittedRecordSubclasses(DatabaseTypeConfig.class, codecs));
   }
 
-  static REv2DatabaseConfig inMemory(boolean enableAccountChangeIndex) {
-    return new InMemory(enableAccountChangeIndex);
+  static DatabaseTypeConfig inMemory(DatabaseConfig databaseConfig) {
+    return new InMemory(databaseConfig);
   }
 
-  static REv2DatabaseConfig rocksDB(String databasePath, boolean enableAccountChangeIndex) {
-    return new RocksDB(databasePath, enableAccountChangeIndex);
+  static DatabaseTypeConfig rocksDB(String databasePath, DatabaseConfig databaseConfig) {
+    return new RocksDB(databasePath, databaseConfig);
   }
 
-  record InMemory(boolean enableAccountChangeIndex) implements REv2DatabaseConfig {}
+  record InMemory(DatabaseConfig databaseConfig) implements DatabaseTypeConfig {}
 
-  record RocksDB(String databasePath, boolean enableAccountChangeIndex)
-      implements REv2DatabaseConfig {}
+  record RocksDB(String databasePath, DatabaseConfig databaseConfig)
+      implements DatabaseTypeConfig {}
```

Comment:
I know, it feels very wrong when looking from Java :). From Rust however it does make sense because it is an enum which gives you the "type" of the database. I want to keep the Rust like it is (maybe change the names for better) because it encodes it in a more idiomatic Rust way (the config for InMemory never has a String passed for path and the config for RocksDB always has).

Maybe renaming `DatabaseTypeConfig` -> `DatabaseConfig`, `DatabaseConfig` -> `DatabaseIndexOptions`/`DatabaseFeatureFlags`?

Reply:
> the config for InMemory never has a String passed for path and the config for RocksDB always has

yeah, I ack this and I respect this; what I meant in my proposal is:

We effectively have a `new Database(DatabaseConfig)` dependency to be constructed, which is very classic and valid 👍 
Using the `DatabaseConfig` we want to specify:
- what database backend to use (i.e. rocksdb vs memory), together with any type-specific details (i.e. file path for rocksdb)
  - let's call this interface `DatabaseBackendConfig`, with impls `RocksdbConfig(Path)` and `MemoryConfig()`
- what features we want from the database layer (i.e. "please store detailed transaction receipt locally")
  - let's call this part `DatabaseFlags`, and it contains fields like `enableLocalTransactionExecutionIndex, enableAccountChangeIndex`

CURRENT STATE:
- `new Database(new RocksdbConfig(databasePath, new DatabaseFlags(..., ...)))`
- `new Database(new MemoryConfig(new DatabaseFlags(..., ...)))`

MY PROPOSAL:
- `new Database(new DatabaseConfig(new RocksdbConfig(databasePath), new DatabaseFlags(..., ...))`
- `new Database(new DatabaseConfig(new MemoryConfig(), new DatabaseFlags(..., ...))`

the point being: `DatabaseFlags` are independent from database backend used, so why would `DatabaseBackendConfig` contain them?

Other possible _syntax_ flavors of my proposal: _(maybe simpler?)_
- `new Database(new DatabaseConfig(new RocksdbConfig(databasePath), ..., ...)` (drop the `DatabaseFlags` grouping, just make them a direct part of `DatabaseConfig`)
- `new Database(new RocksdbConfig(databasePath), new DatabaseFlags(..., ...))` (drop the `DatabaseConfig`, just interpret the "backend type" vs "feature flags" as 2 separate top-level things needed by `Database`)